### DISCUSSION TITLE
How to split a PDF into parts?

### DISCUSSION THREAD

**Discussion by Gowee** [pikepdf/pikepdf#476] [Q&A] [answered]
I am trying to achieve `qpdf --split-pages=...` with pikepdf.

The doc [does mention](https://pikepdf.readthedocs.io/en/latest/topics/pages.html#split-a-pdf-into-single-page-pdfs) the way to split a PDF into single-page PDFs. But it also notes that 

> This example will transfer data associated with each page, so that every page stands on its own. It will not transfer some metadata associated with the PDF as a whole, such as the list of bookmarks.

According to [another note](https://pikepdf.readthedocs.io/en/latest/api/main.html#pikepdf.Pdf.new) in doc

> In most cases, if you are working from an existing PDF, you should open the PDF using [pikepdf.Pdf.open()](https://pikepdf.readthedocs.io/en/latest/api/main.html#pikepdf.Pdf.open) and transform it, instead of a creating a new one, to preserve metadata and structural information. For example, if you want to split a PDF into two parts, you should **open the PDF and transform it into the desired parts**, rather than creating a new PDF and copying pages into it.

How to "transform it into the desired parts"?

**Comment by jbarlow83** [comment, marked as answer]
For example, rather than copying the page to a new PDF, delete all of the pages except the one you want to keep.

**Reply by Gowee** [reply]
Thanks! That sounds reasonable but not intuitive. It would be great if it gets mentioned in the doc.
